src/ble/CHIPBleServiceData.h

```diff
@@ -45,11 +45,15 @@ enum chipBLEServiceDataType
  */
 struct ChipBLEDeviceIdentificationInfo
 {
-    constexpr static uint16_t kDiscriminatorMask     = 0xfff;
-    constexpr static uint8_t kAdditionalDataFlagMask = 0x1;
+    constexpr static uint16_t kDiscriminatorMask           = 0xfff;
+    constexpr static uint8_t kAdditionalDataFlagMask       = 0x1;
+    constexpr static uint8_t kAdvertisementVersionMask     = 0x0f;
+    constexpr static uint8_t AdvertisementVersionShiftBits = 4u;
 
     uint8_t OpCode;
-    uint8_t DeviceDiscriminator[2];
+    // from lsb 0-11 bits (i.e. 8 bits of DeviceDiscriminatorAndAdvVersion[0] and 4 bits of DeviceDiscriminatorAndAdvVersion[1]) for
+    // device discriminator and 12-15 bits (i.e 4 bits of DeviceDiscriminatorAndAdvVersion[1]) for advertisement version
```

Comment:
This is still pretty misleading, because the bits are not "from lsb" inside the bytes....  Probably something more like this:
```suggestion
    // DeviceDiscriminatorAndAdvVersion[0] contains the low 8 bits of the 12-bit discriminator.
    // DeviceDiscriminatorAndAdvVersion[1] contains the high 8 bits of the 12-bit discriminator in its low 4 bits and
    // the 4 bits of the advertisement version in its high 4 bits.
```